BootloaderCommonPkg/Library/TpmLib/TpmLib.c

```diff
@@ -17,11 +17,72 @@
 #include <Library/PcdLib.h>
 #include <Pi/PiBootMode.h>
 #include <IndustryStandard/Tpm2Acpi.h>
+#include <Library/SecureBootLib.h>
 #include "Tpm2CommandLib.h"
 #include "Tpm2DeviceLib.h"
 #include "TpmLibInternal.h"
 #include "TpmEventLog.h"
 
+
+
+/**
+  Get Tcg PCR Hash bank type required by bootloader.
+
+  @retval TpmHashAlg    TCG Algorithm Registry.
+**/
+UINT32
+GetTcgHashEnabled (
+  VOID
+  )
+{
+  UINT32 TpmHashAlg = 0;
+  UINT8  PcdHashType;
+
+  PcdHashType = PcdGet8(PcdMeasuredBootHashType);
+
+  if(PcdHashType == HASH_TYPE_SHA256){

```

Comment:
We have three different enums, HASH_TYPE_SHA256, HASH_ALG_SHA256 and TPM_ALG_SHA256.

I am afraid this will cause bugs later.
Can we atleast get rid of one?

Reply:
Three hashes,
-  TCG Alg registry (bit mask) - Limited to TPM lib
-  TPM Alg registry - Limited to TPM Lib
- Another one is CryptoLib HASH_ALG_TYPE which is used across SBL.

We could think of aligning Crypto HASH_ALG_TYPE with TCG Alg registry. 
I would take this as separate one as it would involves many file changes including data type changes in API's.